### PR TITLE
Connect to Amazon Rekognition

### PR DESCRIPTION
**Pull request by AnilMaktala**
#### Description of changes:
Connect to Amazon Rekognition
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by aspittel — DISMISSED** [review]

File: src/directory/directory.mjs [hunk 1]
```diff
@@ -168,6 +168,11 @@ export const directory = {
                   children: [
                     {
                       path: 'src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-external-ddb-table/index.mdx'
+
+                    },
+                    {
+                      path: 'src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx'
+
```
Comment (timngyn) [inline src/directory/directory.mjs:175]: nit: could we remove the extra newlines?
```suggestion
                    },
                    {
                      path: 'src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx'
```

File: src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+## Step 1 - Set up the project
+
+Set up your project by following the instructions in the [Quickstart guide](/[platform]/start/quickstart/). 
+
+## Step 2 - Install Rekognition Libraries
+We'll create a new API endpoint that'll use the the AWS SDK to call the Amazon Rekognition service. To install the Amazon Rekognition SDK, run the following command in your project's root folder:
+
+```bash
+npm install @aws-sdk/client-rekognition
+
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx:57]:
```suggestion
```

File: src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+export function request(ctx) {
+  return {
+    method: "POST",
+    resourcePath: "/",
+    params: {
+      body: {
+        Image: {
+          S3Object: {
+            Bucket:
+              "<replace with S3 Bucket name>",
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx:133]: Is there a way to set an AppSync environment variable here instead?

File: src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+    .TextDetections.filter((item) => item.Type === "LINE")
+    .map((item) => item.DetectedText)
+    .join("\n")
+    .trim();
+}
+
+
+```
+
+## Step 6 - Define the Custom Query
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx:157]:
```suggestion
## Step 6 - Define the custom query
```

File: src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+```ts title="amplify/data/resource.ts"
+import { type ClientSchema, a, defineData, defineFunction } from '@aws-amplify/backend';
+
+const schema = a.schema({
+  identifyText: a.query()
+    .arguments({
+      path: a.string(),
+    })
+    .returns(a.string())
+    .authorization([a.allow.public()])
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx:170]:
```suggestion
    .authorization(allow => [allow.publicApiKey()])
```

File: src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+    })),
+});
+
+export type Schema = ClientSchema<typeof schema>;
+
+export const data = defineData({
+  schema,
+  authorizationModes: {
+    defaultAuthorizationMode: 'apiKey',
+    // API Key is used for a.allow.public() rules
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx:183]:
```suggestion
```

File: src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+          {textData}
+        </div>
+      </div>
+    </div>
+  );
+}
+
+export default App;
+
+
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx:278]:
```suggestion
```

File: src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx (hunk 3 above)
Comment (AnilMaktala) [reply src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx]: updated code using env variable.

File: src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+  data,
+  storage
+});
+
+const dataStack = Stack.of(backend.data)
+
+// Set environment variables for the S3 Bucket name
+backend.data.resources.cfnResources.cfnGraphqlApi.environmentVariables = {
+  S3_BUCKET_NAME:
+    "<s3-bucket-name>",
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx:97]: can you add the reference here? i.e. `backend.storage.bucket.s3...`

Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-amazon-rekognition/index.mdx:97]:
```suggestion
  S3_BUCKET_NAME: backend.storage.resources.bucket.bucketName,
```

**Review by renebrandel — DISMISSED** [review]

**Review by reesscot — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Pull request merged**